packages/calcite-components/src/tests/commonTests.ts

```diff
@@ -1341,3 +1345,124 @@ export async function t9n(componentTestSetup: ComponentTestSetup): Promise<void>
     await page.waitForChanges();
   }
 }
+
+/**
+ * Helper to test openClose component setup.
+ *
+ * Note that this helper should be used within a `describe` block.
+ *
+ * @example
+ * import { openClose } from "../../tests/commonTests";
+ *
+ * openClose("calcite-tooltip", "open", false, {
+ *   openTooltip: async (page) => {
+ *     await page.keyboard.press("Tab");
+ *     await page.waitForChanges();
+ *   },
+ *   closeTooltip: async (page) => {
+ *     await page.keyboard.press("Tab");
+ *     await page.waitForChanges();
+ *   },
+ * });
+ *
+ * @param {ComponentTestSetup} ComponentTestSetup - A component tag, html, or the tag and e2e page for setting up a test.
+ * @param {string} toggleProp - Toggle property to test. Currently, either "open" or "expanded".
+ * @param {boolean} toggleValue - Indicates the initial value of the toggle property.
+ * @param {{
+ *  openTooltip: (page: E2EPage) => Promise<void>;
+ *  closeTooltip: (page: E2EPage) => Promise<void>;
+ * }} userInputDevice - Optional argument with functions to simulate user input (mouse or keyboard), to open or close the component.
+ */
+
+export function openClose(
+  componentTestSetup: ComponentTestSetup,
+  toggleProp: string,
+  toggleValue: boolean,
+  userInputDevice?: {
```

Comment:
Can you look at introducing an `options` object? If you look at other test utils, additional props not required for the test usually go in an `options` argument.